liblumen_alloc/src/erts/term/map.rs

```diff
@@ -47,6 +47,13 @@ impl Map {
         self.value.contains_key(&key)
     }
 
+    pub fn keys(&self) -> Vec<Term> {
+        let mut key_vec: Vec<Term> = Vec::new();
+        key_vec.extend(self.value.keys());
+
+        key_vec
```

Comment:
This entire function can be simplified into:

```rust
pub fn keys(&self) -> Vec<Term> {
  self.keys().into()
}
```

Reply:
@ShaneWilton switch to @archseer's recommendation.

Reply:
Thanks for the suggestion! I tried making this change, and it looks like trait isn't implemented for the types in question:

```
51 |         self.value.keys().into()
   |                           ^^^^ the trait `std::convert::From<hashbrown::map::Keys<'_, erts::term::term::Term, erts::term::term::Term>>` is not implemented for `std::vec::Vec<erts::term::term::Term>`
   |
   = help: the following implementations were found:
             <std::vec::Vec<T> as std::convert::From<&[T]>>
             <std::vec::Vec<T> as std::convert::From<&mut [T]>>
             <std::vec::Vec<T> as std::convert::From<std::borrow::Cow<'a, [T]>>>
             <std::vec::Vec<T> as std::convert::From<std::boxed::Box<[T]>>>
           and 5 others
```

Reply:
Hmm, seems like hash brown's traits don't match the std ones. Try: self.value.keys().into_iter().into(), I think From<Iterator> is implemented for vec.

Reply:
I'll give that a try when I'm back at a computer! Thanks for your patience with my questions, I really appreciate the help :)